src/gui_common/SpeciesPreview.cs

```diff
@@ -0,0 +1,50 @@
+﻿using Godot;
+
+public class SpeciesPreview : Control
+{
+    [Export]
+    public NodePath TextureRectPath = null!;
+
+    private TextureRect textureRect = null!;
+    private Species? previewSpecies;
+    private ImageTask? task;
+
+    public Species? PreviewSpecies
+    {
+        get => previewSpecies;
+        set
+        {
+            previewSpecies = value;
+
+            if (previewSpecies != null)
+                UpdatePreviewSpecies();
+        }
+    }
+
+    public override void _Ready()
+    {
+        base._Ready();
+
+        textureRect = GetNode<TextureRect>(TextureRectPath);
+    }
+
+    public override void _Process(float delta)
+    {
+        base._Process(delta);
+
+        if (task?.Finished == true)
+        {
+            textureRect.Texture = task.FinalImage;
+            task = null;
+        }
+    }
+
+    private void UpdatePreviewSpecies()
+    {
+        if (previewSpecies is MicrobeSpecies microbeSpecies)
+        {
+            task = new ImageTask(microbeSpecies);
+            PhotoStudio.Instance.SubmitTask(task);
+        }
```

Comment:
I'd add an else here with an error print that an unknown species type was attempted to be previewed.